client/src/app/toolbar/toolbar-menu/toolbar-menu.component.spec.ts

```diff
@@ -52,12 +53,30 @@ describe('ToolbarMenuComponent', () => {
     userService = TestBed.get(UserService);
 
     spyOn(authService, 'linkOrSignInWithGitHub').and.returnValue(of(dummyUser));
+    spyOn(authService, 'signInWithGitHub').and.returnValue(of(dummyUser));
     spyOn(authService, 'signOut').and.returnValue(of(null));
     spyOn(userService, 'createUserIfMissing').and.returnValue(of(dummyUser));
   });
 
+  it('should signup via GitHub', () => {
+    spyOn(userService, 'observeUserChanges').and.returnValue(of(dummyUser));
+    spyOn(userService, 'getCurrentUserSaved').and.returnValue(of(false));
+    fixture.detectChanges();
+    const menuButton = fixture.debugElement.query(By.css('[mat-icon-button]'));
+
+    // open menu
+    menuButton.triggerEventHandler('click', null);
+
+    const loginButton = fixture.debugElement.query(By.css('.mat-menu-panel button'));
+
+    loginButton.triggerEventHandler('click', null);
+    expect(userService.getCurrentUserSaved).toHaveBeenCalled();
+    expect(authService.signInWithGitHub).toHaveBeenCalled();
+  });
```

Comment:
Super happy to see your PR comes with tests as well! I appreciate the effort. However this test is pretty much the same as the one below, just that `getCurrentUserSaved()` is mocked to return a different value, but we only check for `toHaveBeenCalled()` anyways.

I think it's fine to keep the change in the test below, but get rid off this one.